src/index.ts

```diff
@@ -428,7 +435,7 @@ class NixInstallerAction {
       extraConf += `access-tokens = ${serverUrl}=${this.githubToken}`;
       extraConf += "\n";
     }
-    if (this.trustRunnerUser !== null) {
+    if (this.trustRunnerUser !== null && this.trustRunnerUser) {
```

Comment:
`getBool` returns a Boolean, so you can update the type of `trustRunnerUser` to `boolean` rather that `boolean | null` and thus you don't need the null check here.